drivers/i2s/i2s_sam_ssc.c

```diff
@@ -960,28 +962,17 @@ static struct device *get_dev_from_dma_channel(u32_t dma_channel)
 
 static void i2s0_sam_irq_config(void)
 {
-	IRQ_CONNECT(SSC_IRQn, CONFIG_I2S_SAM_SSC_0_IRQ_PRI, i2s_sam_isr,
+	IRQ_CONNECT(DT_INST_IRQN(0), DT_INST_IRQ(0, priority), i2s_sam_isr,
 		    DEVICE_GET(i2s0_sam), 0);
 }
 
-static const struct soc_gpio_pin i2s0_pins[] = {
-	PIN_SSC0_TK,
-	PIN_SSC0_TF,
-	PIN_SSC0_TD,
-#ifdef CONFIG_I2S_SAM_SSC_0_PIN_RK_EN
-	PIN_SSC0_RK,
-#endif
-#ifdef CONFIG_I2S_SAM_SSC_0_PIN_RF_EN
-	PIN_SSC0_RF,
-#endif
-	PIN_SSC0_RD,
-};
+static const struct soc_gpio_pin i2s0_pins[] = ATMEL_SAM_DT_PINS(0);
```

Comment:
It's going to complicate implementation, however it's a fairly common configuration where RK, RF pins are not connected.

![Screenshot from 2020-05-05 21-12-52](https://user-images.githubusercontent.com/28196316/81106199-74824100-8f15-11ea-8589-3acfdac9c26e.png)

It seems like at the moment there is no possibility to have RK, RF signals disconnected.

Reply:
I underestimated the new DT generation scheme. There is no need to change the implementation. It's enough that user defines their own `pinctrl-0` property for the `ssc` node.

The one issue that remains is if we should provide 'default' `pinctrl-0` property at the SoC level.

Atmel is a bit special, its pinmux is fairly rigid. In any given peripheral few pins, if any, can be assigned to an alternative location. But these choices exist and it feels improper to provide default configuration at the SoC level. Pin assignments are board specific and as soon as there exist a choice any default is wrong. Connecting incorrect signal to a pin can potentially damage the board.

At the same time if a peripheral doesn't allow to connect any of its pins to an alternate location forcing the user to define `pinctrl-0` property at the board level also feels like an overkill.

Reply:
I tried to provide `pinctrl-0` in the SoC dtsi files in the cases that either make sense or if there is only one option for a given peripheral.  I'm find if because of how SSC is used if we want to move the `pinctrl-0` setting to the board file instead.